gen-resourcesdocs/pkg/outputs/kwebsite/section.go

```diff
@@ -19,7 +21,22 @@ type Section struct {
 // AddContent adds content to a section
 func (o Section) AddContent(s string) error {
 	i := len(o.chapter.data.Sections)
-	o.chapter.data.Sections[i-1].Description = s
+
+	// Regular expression to find code blocks
+	re := regexp.MustCompile("```(?s)(.*?)```")
+
+	// Replace code blocks with properly formatted HTML
+	processed := re.ReplaceAllStringFunc(s, func(match string) string {
+		// Remove the backticks and trim whitespace
+		code := strings.TrimSpace(strings.TrimPrefix(strings.TrimSuffix(match, "```"), "```"))
+		// Escape HTML special characters
+		code = html.EscapeString(code)
+		// Preserve line breaks
+		code = strings.ReplaceAll(code, "\n", "<br>")
+		return "<pre><code>" + code + "</code></pre>"
+	})
```

Comment:
I'm move inclined to treat the description text as markdown.

Reply:
Thank you for the markdown suggestion @tengqm - could you point me to any examples of this approach in this repo or recommend any Go libraries for implementation? I noticed that [russross/blackfriday](https://github.com/russross/blackfriday) is installed, that would be my first thought but I am open to suggestions.

Reply:
We use [goldmark](https://github.com/yuin/goldmark) in other generators.

Reply:
Thanks for the guidance @tengqm, I have replaced the html specific rendering logic with a goldmark Parser.